### ISSUE TITLE
groovy.lang.MissingMethodException: No signature of method: java.util.Date.format()

### ISSUE THREAD

**Issue by ekc** [groovy/groovy-eclipse#1383]
# Environment

- Eclipse: 2022-06 (4.24.0) Build id: 20220609-1112
- Eclipse Groovy Development Tools	4.6.0.v202206301445-e2206-RELEASE

# Problem
I tried to invoke the format(String) method on the Date object within Eclipse Groovy Development Tools. Unfortunately, groovy.lang.MissingMethodException is thrown.
However, the same code works fine in the GroovyConsole.
## Steps to reproduce
Run As -> Java Application of the code below,
```groovy
package ekc.selfstudy05

class DateDemo01 {
	static void main(args) {
		def today = new Date().format("MMdd")
		println today
	}
}
```
gives the following error ...
```
Exception in thread "main" groovy.lang.MissingMethodException: No signature of method: java.util.Date.format() is applicable for argument types: (String) values: [MMdd]
Possible solutions: from(java.time.Instant), stream(), toYear(), getAt(java.lang.String), parse(java.lang.String), print(java.io.PrintWriter)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:72)
	at org.codehaus.groovy.vmplugin.v8.IndyGuardsFiltersAndSignatures.unwrap(IndyGuardsFiltersAndSignatures.java:161)
	at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache(IndyInterface.java:318)
	at ekc.selfstudy05.DateDemo01.main(DateDemo01.groovy:6)
```

**Comment by eric-milles** [comment]
The extension methods for `java.util.Date` have moved to the `groovy-dateutil` module.  You can add this to your classpath in a number of ways.  The simplest for a quick test is to add `@Grab('org.codehaus.groovy:groovy-dateutil:2.5.18')` to the top of your script.  You can adjust the version if you are referencing a different Groovy version.

http://groovy-lang.org/releasenotes/groovy-2.5.html#Groovy2.5releasenotes-Breakingchanges
https://stackoverflow.com/questions/57548284/grails-4-groovys-date-enhancement-methods-missing

**Comment by ekc** [comment]
Hello @eric-milles,

1. Annotating `@Grab(group='org.apache.groovy', module='groovy-datetime', version='4.0.3')` gives the following RunTimeException
```
Exception in thread "main" java.lang.ExceptionInInitializerError
Caused by: java.lang.RuntimeException: No suitable ClassLoader found for grab
	at groovy.grape.GrapeIvy.chooseClassLoader(GrapeIvy.groovy:179)
	at groovy.grape.GrapeIvy.grab(GrapeIvy.groovy:249)
	at groovy.grape.Grape$1.run(Grape.java:172)
	at groovy.grape.Grape$1.run(Grape.java:158)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at groovy.grape.Grape.grab(Grape.java:158)
	at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache(IndyInterface.java:318)
	at ekc.selfstudy05.DateDemo01.<clinit>(DateDemo01.groovy)
```
Can you please point me to a document to configure `@Grab` to use an appropriate Classloader to load Groovy `java.util.Date`?

2. The same code (without `@Grab`) works fine with Groovy command-line, GroovyConsole, and Intellij idea

**Comment by ekc** [comment]
Many thanks @eric-milles.
I resolved the problem by adding groovy-dateutil-4.0.3.jar to Groovy Libraries classpath with the following steps
1. Download groovy-dateutil-4.0.3.jar from [https://mvnrepository.com/artifact/org.apache.groovy/groovy-dateutil/4.0.3](https://mvnrepository.com/artifact/org.apache.groovy/groovy-dateutil/4.0.3) and put it into ~/.groovy/lib
2. From the Package Explorer view of the Groovy project, choose the "Groovy Libraries" Classpath and use the context menu "Build Path -> Configure Build Path..."
3. Select "Edit..." button on the right hand side
4. Under the "User libraries" section,
Should jars in the ~/.groovy/lib directory be included on the classpath?
yes
5. Select Finish
